uint64 inc
uint8 SX_Red
uint8 SX_Green
uint8 SX_Blue
uint8 DX_Red
uint8 DX_Green
uint8 DX_Blue